cockatrice/src/player.cpp

```diff
@@ -2602,6 +2615,90 @@ void Player::actSetPT()
     game->sendGameCommand(prepareGameCommand(commandList), playerid);
 }
 
+void Player::actChangePT() // backwards compatible with old "set" flow
+{
+    QString oldPT;
+    int playerid = id;
+
+    auto sel = scene()->selectedItems();
+    for (const auto &item : sel) {
+        auto *card = static_cast<CardItem *>(item);
+        if (!card->getPT().isEmpty()) {
+            oldPT = card->getPT();
+        }
+    }
+    bool ok;
+    dialogSemaphore = true;
+    QString pt = QInputDialog::getText(nullptr, tr("Change power/toughness"), tr("Please enter the change to PT:"),
+                                       QLineEdit::Normal, oldPT, &ok);
+    dialogSemaphore = false;
+    if (clearCardsToDelete()) {
+        return;
+    }
+    if (!ok) {
+        return;
+    }
+
+    bool addPow, addTou;
+    int pow, tou;
+    {
+        int sep = pt.indexOf('/');
+        QString p1 = pt.left(sep);
+        QString p2 = pt.mid(sep + 1);
```

Comment:
This lacks proper user input validation. Just enter a single "/" in the dialog and it will crash.

Reply:
oh wow, you're right, I could have seen that immediately, I don't see where it actually checks that in the old version though.

Reply:
oh, I see, if isEmpty() return, why would I remove that while half asleep?

Reply:
actually the crash completely unrelated and is caused when drawing, and while fixing this I'll actually fix #3544 :)